### ISSUE TITLE
Race condition in accept loop when stats are enabled

### ISSUE THREAD

**Issue by jshiell** [cherrypy/cheroot#198]
❓ **I'm submitting a ...**
- [X] 🐞 bug report
- [ ] 🐣 feature request
- [ ] ❓ question about the decisions made in the repository

🐞 **Describe the bug. What is the current behavior?**

When starting a server with monitoring enabled, there seems to be an intermittent race condition where the start time is accessed before it has been initialised, resulting in an exception.

❓ **What is the motivation / use case for changing the behavior?**

Fewer exceptions :-)

💡 **To Reproduce**

Toggle stats reporting on while a request is in progress, e.g. starting the server while an uptime monitoring query is being performed.

💡 **Expected behavior**

Stats are recorded.

📋 **Details**

Example stack trace:
```python-traceback
[03/May/2019:14:29:50] ENGINE Bus STARTING
[03/May/2019:14:29:50] ENGINE Started monitor thread 'Autoreloader'.
[03/May/2019:14:29:50] ENGINE Serving on http://0.0.0.0:50001
[03/May/2019:14:29:50] ENGINE Bus STARTED
Exception in thread CP Server Thread-3:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/path/to/project/venv/lib/python3.7/site-packages/cheroot/workers/threadpool.py", line 121, in run
    self.work_time += time.time() - self.start_time
TypeError: unsupported operand type(s) for -: 'float' and 'NoneType'
```

📋 **Environment**

- Cheroot version: 6.5.5
- CherryPy version: 18.1.1
- Python version: 3.7.3
- OS: OS X 10.14.4
- Browser: all

📋 **Additional context**

I believe this is probably occurring as [`threadpool.py`](https://github.com/cherrypy/cheroot/blob/25c3a277f311739f85980493f19123ceb2731939/cheroot/workers/threadpool.py#L111) does two independent checks on `self.server.stats['Enabled']` - one to initialise the start time, one to record the results. If the value of `self.server.stats['Enabled']` changes while a request is in progress then we can assume there's a chance that we'll try the result block without the initialising the start time.

**Comment by jshiell** [comment]
Bump.

**Comment by webknjaz** [comment]
Hey @jshiell, do you have a reproducer script for this?

**Comment by jshiell** [comment]
I'm afraid not. We originally saw it with our functional tests, which will start the server as part of test init. The rest was deduction based on the stack trace. Given there was a pretty clear cause I didn't think more investigation was warranted.

**Comment by webknjaz** [comment]
Well, in general, it's useful to collect as much info as available: it helps to get back to the issue after some delay because free time is a luxury plus it's useful to people who use search while facing something similar to land here. It's beneficial to have a stable reproducer so that we could verify fixes too.
One of the ways to demo such a problem is to submit a test which is failing and then temporary `xfail` it so that when somebody will try fixing it, they'd just remove that `xfail` mark in order to verify the solution and that test would stay there also to prevent regressions.

@jshiell does this occur constantly? Do you toggle stats dynamically in your tests?

**Comment by webknjaz** [comment]
@jshiell I think 9f7affe should handle this.

**Comment by jshiell** [comment]
For the record; no - and no, we don't toggle. It just seemed to be a timing issue on startup. It looks like we fixed it by toggling thread order to give the stats reporting thread a little longer to startup.

That commit looks perfect - thanks very much @webknjaz, much obliged!

**Comment by webknjaz** [comment]
Released the patch in [v6.5.7](https://pypi.org/project/cheroot/6.5.7/).
